README.md
# Polaric webapp

The "Polaric Server" is mainly a web based service to present APRS tracking information on maps and where the information is updated in real-time. It is originally targeted for use by radio amateurs in voluntary search and rescue service in Norway. It consists of a web application and a server program (APRS daemon). 

The web application lets the user browse maps. It can present live APRS information (from the aprsd backend) on top of the map. Users can update information, add objects, etc.. The application supports use of the Mapcache Apache module that can render and/or cache maps. Web clients can also be set up to use external map sources directly. 

More documentation on the project can be found here: 
http://aprs.no/polaricserver

A rewrite/port to OpenLayers 4+ is in progress: 
https://github.com/PolaricServer/webapp2


## System requirements

Linux/Apache/PHP platform (tested with Debian/Ubuntu/Mint).

We support automatic installation packages for Debian systems. It shouldn't be too hard to port it to e.g. Windows if anyone wants to do the job. 
 
## Installation

We provide a Debian package (debian Jessie or later) For information on getting 
started on a Debian platform (or derivative) please see: 
http://aprs.no/dokuwiki?id=install.dev

More documentation on the project can be found here: 
http://aprs.no/polaricserver

## Building from source 

Build from the source is done by a plain old makefile. Yes I know :)
Setup for generating Debian packages is included. You may use the 'debuild' 
command.

You will need the Closure Compiler from Google to compress the javascript 
code. See compile-js.sh. 



debian/changelog
polaric-webapp (2.1.1) stable; urgency=medium

  * Bugfix 

 -- Øyvind Hanssen <ohanssen@acm.org>  Sat, 02 Dec 2017 19:33:45 +0100

polaric-webapp (2.1) stable; urgency=medium

  *  Fixes and modifications (see changelog of aprsd). 

 -- Øyvind Hanssen <ohanssen@acm.org>  Thu, 16 Nov 2017 21:46:39 +0100

polaric-webapp (2.0.3) stable; urgency=medium

  * Offline/online mode 

 -- Øyvind Hanssen <ohanssen@acm.org>  Sun, 11 Dec 2016 14:01:57 +0100

polaric-webapp (2.0.2) stable; urgency=medium

  * Small fix 

 -- Øyvind Hanssen <ohanssen@acm.org>  Sat, 10 Dec 2016 01:24:37 +0100

polaric-webapp (2.0.1) stable; urgency=medium

  * Fix php dependencies, use address service, etc..  

 -- Øyvind Hanssen <ohanssen@acm.org>  Thu, 08 Dec 2016 23:55:09 +0100

polaric-webapp (2.0) stable; urgency=medium

  * Websockets, etc.. 
  * Fixes and additions... 

 -- Øyvind Hanssen <ohanssen@acm.org>  Mon, 18 Apr 2016 21:04:48 +0200

polaric-webapp (1.9) unstable; urgency=medium

  * Some fixes (see changelog of aprsd). 

 -- Øyvind Hanssen <ohanssen@acm.org>  Wed, 30 Dec 2015 00:05:40 +0100

polaric-webapp (1.8) unstable; urgency=medium

  * Fixes and modifications (see changelog of aprsd). 

 -- Øyvind Hanssen <ohanssen@acm.org>  Sat, 15 Aug 2015 23:23:41 +0200

polaric-webapp (1.7) unstable; urgency=medium

  * Fixes and modifications (see changelog of aprsd)

 -- Øyvind Hanssen <ohanssen@acm.org>  Sat, 21 Feb 2015 23:26:31 +0000

polaric-webapp (1.6) unstable; urgency=low

  * Fixes and modifications (see changelog of aprsd)

 -- Øyvind Hanssen <ohanssen@acm.org>  Thu, 27 Nov 2014 21:03:29 +0100

polaric-webapp (1.5) unstable; urgency=medium

  * Fixes and modifications (see changelog of aprsd) 

 -- Øyvind Hanssen <ohanssen@acm.org>  Tue, 02 Sep 2014 22:39:39 +0200

polaric-webapp (1.4) unstable; urgency=low

  * Some fixes and modifications (see changelog of aprsd) 

 -- Øyvind Hanssen <ohanssen@acm.org>  Mon, 10 Mar 2014 23:06:45 +0000

polaric-webapp (1.3) unstable; urgency=low

  * Several fixes (see changellog of aprsd) 

 -- Øyvind hanssen <ohanssen@acm.org>  Thu, 02 Jan 2014 00:47:26 +0100

polaric-webapp (1.2) unstable; urgency=low

  * Several fixes (see changelog of aprsd) 

 -- Øyvind hanssen <ohanssen@acm.org>  Sat, 27 Apr 2013 13:38:49 +0200

polaric-webapp (1.1) unstable; urgency=low

  * Several fixes, performance improvement, release.. 

 -- Øyvind hanssen <ohanssen@acm.org>  Mon, 14 Jan 2013 23:34:50 +0100

polaric-webapp (1.0.99+1.1-beta3) unstable; urgency=low

  * Fixes to API. Move to git, etc.. 

 -- Øyvind hanssen <ohanssen@acm.org>  Fri, 28 Dec 2012 01:09:33 +0100

polaric-webapp (1.0.99+1.1-beta2) unstable; urgency=low

  * Ready for deploying on aprs.no 

 -- Øyvind hanssen <ohanssen@acm.org>  Wed, 24 Oct 2012 21:42:49 +0200

polaric-webapp (1.0.99+1.1-beta1) unstable; urgency=low

  * Stabilize for release. 

 -- Øyvind hanssen <ohanssen@acm.org>  Mon, 15 Oct 2012 00:12:09 +0200




index.php
<!DOCTYPE html>

<?php 

// If browser has been restarted or somehow the auth info has gone, 
// we nee to delete the tryLogin cookie (see also login.php). 
 if (!isset($_SERVER['PHP_AUTH_USER']))
       setcookie("polaric.tryLogin", "", time()-3600);

$isMobile = false; 
$isIframe = false; 
$isDebug = false;

// Try to detect if mobile browser or app is in to be used in an iframe
$useragent=$_SERVER['HTTP_USER_AGENT'];
if((isset($_GET['mobile']) &&  preg_match('/true|TRUE/i', $_GET['mobile'])) || 
   preg_match('/android.+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i',$useragent)||preg_match('/1207|6310|6590|3gso|4thp|50[1-6]i|770s|802s|a wa|abac|ac(er|oo|s\-)|ai(ko|rn)|al(av|ca|co)|amoi|an(ex|ny|yw)|aptu|ar(ch|go)|as(te|us)|attw|au(di|\-m|r |s )|avan|be(ck|ll|nq)|bi(lb|rd)|bl(ac|az)|br(e|v)w|bumb|bw\-(n|u)|c55\/|capi|ccwa|cdm\-|cell|chtm|cldc|cmd\-|co(mp|nd)|craw|da(it|ll|ng)|dbte|dc\-s|devi|dica|dmob|do(c|p)o|ds(12|\-d)|el(49|ai)|em(l2|ul)|er(ic|k0)|esl8|ez([4-7]0|os|wa|ze)|fetc|fly(\-|_)|g1 u|g560|gene|gf\-5|g\-mo|go(\.w|od)|gr(ad|un)|haie|hcit|hd\-(m|p|t)|hei\-|hi(pt|ta)|hp( i|ip)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)
|50|54|e\-|e\/|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(di|rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|xda(\-|2|g)|yas\-|your|zeto|zte\-/i',substr($useragent,0,4)))
  $isMobile = true;
  
if ((isset($_GET['iframe']) && preg_match('/true|TRUE/i', $_GET['iframe'])))
  $isIframe = true;
  
if ((isset($_GET['debug']) && preg_match('/true|TRUE/i', $_GET['debug'])))
  $isDebug = true;
  
include_once( 'webappconfig.php' ); 
?>

<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="viewport" content="user-scalable=no, width=device-width, height=device-height">

<title><?php echo($siteTitle); ?></title>
<link rel="icon" href="favicon.ico" type="image/x-icon">
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon"> 
<link rel="stylesheet" href="jquery-ui/css/smoothness/jquery-ui.min.css" type="text/css">
<link rel="stylesheet" href="OpenLayers/theme/default/style.css" type="text/css">
<link rel="stylesheet" href="KaMap/scalebar/scalebar-fat.css" type="text/css"  >
<link rel="stylesheet" href="style/screen.css" type="text/css">
<link rel="stylesheet" href="style/tools.css"  type="text/css">
<link rel="stylesheet" href="style/popup.css"  type="text/css">
<link rel="stylesheet" href="style/aprs.css"   type="text/css">
<?php 
   if ($isMobile)
       echo('<link rel="stylesheet" href="style/mobil.css" type="text/css">');
   else if ($isIframe)
       echo('<link rel="stylesheet" href="style/iframe.css" type="text/css">');
       
       
   /* Auto include of CSS files */
   if ($handle = opendir('/etc/polaric-webapp/www/auto')) {
       while (false !== ($file = readdir($handle)))
          if ($file != "." && $file != ".." && strtolower(substr($file, strrpos($file, '.') + 1)) == 'css')
              echo('<link rel="stylesheet" href="config/auto/'.$file.'" type="text/css">');
       closedir($handle);
    }


  // Add extra CSS stylesheets
  foreach ($cssIncludes as $inc) 
     echo ('<link rel="stylesheet" href="config/'.$inc.'" type="text/css">');
  
  // Add content of headinclude file
  $headInclude = "/etc/polaric-webapp/".$headInclude;
  if ($headInclude != "" && file_exists($headInclude) ) {
     $fh = fopen($headInclude, 'r');
     while (!feof($fh)) 
       echo fgets($fh);
     fclose($fh);
  }
?>

</head>
<body onload="<?php if ($isMobile) echo("myOnLoad_mobile();");
                    else if ($isIframe) echo ("myOnLoad_iframe();");
                    else echo("myOnLoad();"); ?>">
<div id="PolaricVP">
<?php
  if ($isMobile || $isIframe)
    include_once('form_small.php');
  else
    include_once('form_full.php');
?>

<img id="toolbarToggler" onclick="toggleToolbar(this);" alt="toggle toolbar" src="KaMap/images/a_pixel.gif">
 
<?php
  if ($creditLogo!= ""  && !$isMobile && !$isIframe) 
      echo ('<img id="clogo" src="'.$creditLogo.'">');
?>


<!-- reference information -->
<img id="refToggler" onclick="toggleReference(this);" alt="toggle reference" src="KaMap/images/a_pixel.gif">
<div id="reference">
    <div id="scaleReference">
        <div id="scaleBackground" class="transparentBackground"></div>
        <div id="scalebar"></div>
        <div id="scale">current scale</div>
    </div>
    <div id="geoPosition"></div>
</div>    
</div>

<?php
  if (!$isMobile) 
     echo('<script type="text/javascript" src="OpenLayers/lib/Firebug/firebug.js"></script>');
?>
<script type="text/javascript" src="i18n/compiled.js"></script>
<script type="text/javascript" src="jquery/jquery.min.js"></script>
<script type="text/javascript" src="jquery-ui/jquery-ui.min.js"></script>
<script type="text/javascript" src="OpenLayers/openlayers-polaric.js"></script>
<script type="text/javascript" src="proj4js-compressed.js"></script>
<script type="text/javascript" src="Aprs/configSupport.js"></script>
<script type="text/javascript" src="mapconfig.js"></script>


    
<?php
  /******************************
   * Auto include of JS files   *
   * for config                 *
   ******************************/
  if ($handle = opendir('/etc/polaric-webapp/www/auto_config')) {
     while (false !== ($file = readdir($handle)))
        if ($file != "." && $file != ".." && strtolower(substr($file, strrpos($file, '.') + 1)) == 'js')
            echo('<script type="text/javascript" src="config/auto_config/'.$file.'" ></script>');
     closedir($handle);
  }


  if  (!$isDebug) {
     echo('<script src="KaMap/kamap-core.js"></script>');
     echo('<script src="XMLOverlay/compiled.js"></script>');
     echo('<script src="Aprs/compiled.js"></script>');
  }
  else {
     /************************************************************************
      * Debug mode, use source javascript files instead of compiled files
      ************************************************************************/
      
     echo('<script src="KaMap/eventManager.js"></script>');
     echo('<script src="KaMap/DHTMLapi.js"></script>');
     echo('<script src="KaMap/xhr.js"></script>');
     echo('<script src="KaMap/touchHandler.js"></script>');
     echo('<script src="KaMap/kaMap.js"></script>');
     echo('<script src="KaMap/kaTool.js"></script>');
     echo('<script src="KaMap/kaQuery.js"></script>');
     echo('<script src="KaMap/kaMouseTracker.js"></script>');
     echo('<script src="KaMap/scalebar/scalebar.js"></script>');
     echo('<script src="XMLOverlay/labelStyle.js"></script>');    
     echo('<script src="XMLOverlay/kaXmlOverlay.js"></script>');
     echo('<script src="XMLOverlay/kaXmlFeature.js"></script>');
     echo('<script src="XMLOverlay/kaXmlIcon.js"></script>');
     echo('<script src="XMLOverlay/kaXmlLabel.js"></script>');
     echo('<script src="XMLOverlay/kaXmlPoint.js"></script>');
     echo('<script src="XMLOverlay/kaXmlSymbol.js"></script>');
     echo('<script src="XMLOverlay/labelStyle.js"></script>');
     echo('<script src="Aprs/js.cookie.js"></script>');
     echo('<script src="Aprs/statkartWPS.js"></script>');
     echo('<script src="Aprs/statkartName.js"></script>');
     echo('<script src="Aprs/WXreport.js"></script>');
     echo('<script src="Aprs/api.js"></script>');
     echo('<script src="Aprs/iscroll.js"></script>');
     echo('<script src="Aprs/popup.js"></script>');  
     echo('<script src="Aprs/contextmenu.js"></script>');  
     echo('<script src="Aprs/layerSwitcher.js"></script>'); 
     echo('<script src="Aprs/mapupdate.js"></script>');      
     echo('<script src="Aprs/messages.js"></script>');    
     echo('<script src="Aprs/menu.js"></script>');
     echo('<script src="Aprs/jscoord.js"></script>');
     echo('<script src="Aprs/filterProfile.js"></script>');
     echo('<script src="Aprs/auth.js"></script>');
     echo('<script src="Aprs/startUp.js"></script>');
  }
?>
<script src="KaMap/tools/kaRubberZoom.js"></script>
<script src="KaMap/tools/myKaRuler.js"></script> 
       
<?php

   /******************************
    * Auto include of JS files   *
    ******************************/
   if ($handle = opendir('/etc/polaric-webapp/www/auto')) {
       while (false !== ($file = readdir($handle)))
          if ($file != "." && $file != ".." && strtolower(substr($file, strrpos($file, '.') + 1)) == 'js')
              echo('<script type="text/javascript" src="config/auto/'.$file.'" ></script>');
       closedir($handle);
    }

  /********************************
   * Add extra Javascript files   *
   ********************************/
  foreach ($jsIncludes as $inc) 
    if ($inc != "." && $inc != ".." && strtolower(substr($inc, strrpos($inc, '.') + 1)) == 'js')
       echo ('<script type="text/javascript" src="config/'.$inc.'" ></script>');
?>
</body>
</html>
